replication_manager5.sh

```diff
@@ -0,0 +1,503 @@
+                        # Looks like we cannot reach the master, let's try to reconnect
+                        
+                        masterOk=$(try_masters $remoteCluster)
+                        if [ "$masterOk" -eq 1 ]; then
+                            # we succeeded reconnecting
+                            $MYSQL -e "
+                            update percona.replication 
+                             set isSlave='Yes', localIndex=$wsrep_local_index, lastHeartbeat=now() 
+                             where connectionName = '${wsrep_cluster_name}-${remoteCluster}' 
+                               and host = '$wsrep_node_name'"         
+                        else
+                            # We failed, bailing-out
+                            bail_out Failed
+                            send_email "Node $wsrep_node_name failed as a slave for the cluster $wsrep_cluster_name, IO thread not running" "Failed slave"
+                        fi
+                    fi
+                fi
+                
+                # Sanity check, is there more than one slave reporting for the connection?
+                slaveCount=$($MYSQL -BN -e "select count(*) from percona.replication where isSlave = 'Yes' and connectionName = '${wsrep_cluster_name}-${remoteCluster}' and unix_timestamp(lastHeartbeat) > unix_timestamp() - $FAILED_REPLICATION_TIMEOUT")
+                if [ "$slaveCount" -gt 1 ]; then
+                    # that's bad, more than one slave for the cluster... bailout
+                    bail_out No
+                    
+                    send_email "Two nodes were slaves for the cluster $wsrep_cluster_name, stopping slave on node $wsrep_node_name" "Two slaves"
+                    
+                fi
+                
+            elif [ "$myState" == "No" ]; then
+                # We are not defined as a slave in the cluster but we are... bailout
+                bail_out No
+            elif [ "$myState" == "Failed" ]; then
+                # We have failed and we are a slave, this is abnormal, fix state to 'No' and bailout
+                bail_out No
+            elif [ "$myState" == "Proposed" ]; then
+                # Sanity cleanup, if the node is a slave is still at Proposed, need to be updated
+                $MYSQL -e "
+                update percona.replication 
+                set isSlave='Yes', localIndex=$wsrep_local_index, lastHeartbeat=now() 
+                where connectionName = '${wsrep_cluster_name}-${remoteCluster}' 
+                  and host = '$wsrep_node_name'"  
+            fi
+        fi
+    done
+else
+    # cluster node is not sane for this node
+
+    if [ "a$Master_Host" != "a" ]; then 
+        # This node is currently a slave but not in the primary group, bailing out
+        bail_out No
+        send_email "Node $wsrep_node_name is not longer part of the cluster $wsrep_cluster_name, stopping replication" "Failed slave"
```

Comment:
do we really need to send emails? it will require sendmail be in dependencies....

Reply:
I have add a parameter to control this by default NO, we will not send it. 
but no dependency is not needed. If we do not have a mailer we will automatically not send it, period.